src/server/handlers/launchpad.js

```diff
@@ -435,6 +437,33 @@ export const findSnap = (req, res) => {
     .catch((error) => sendError(res, error));
 };
 
+const memcachedDel = (cachedId) => {
+  return new Promise((resolve, reject) => {
+    getMemcached().del(cachedId, (err) => {
+      if (err) {
+        logger.error(`Error deleting ${cachedId} from memcached:`, err);
+        reject(err);
+      }
+      resolve();
+    });
+  });
+};
+
+const clearSnapCache = (repositoryUrl) => {
+  const repository = parseGitHubUrl(repositoryUrl);
+  const enabledReposCacheId = getUrlPrefixCacheId(getRepoUrlPrefix(repository.owner));
+  const snapCacheId = getRepositoryUrlCacheId(repositoryUrl);
+  const snapNameCacheId = getSnapNameCacheId(repositoryUrl);
+
+  logger.info(`Clearing caches for ${repositoryUrl}: ${enabledReposCacheId}, ${snapCacheId}, ${snapNameCacheId}`);
+
+  return Promise.all([
+    memcachedDel(enabledReposCacheId),
+    memcachedDel(snapCacheId),
+    memcachedDel(snapNameCacheId)
+  ]);
```

Comment:
I think snap names are cached twice, once on their own and again with the rest of the snap data. Is it worth consolidating these cache items?